src/resources/ResourceSnapshots/ResourceSnapshotsInterfaces.ts

```diff
@@ -161,7 +162,41 @@ export interface ResourceSnapshotsReportOperationResult {
 
 export type ResourceSnapshotsReportOperationResults = Record<string, ResourceSnapshotsReportOperationResult[]>;
 
+export enum SnapshotsReportStages {
+    COMPUTING_OPERATIONS = 'COMPUTING_OPERATIONS',
+    CREATING_SNAPSHOT = 'CREATING_SNAPSHOT',
+    INITIAL_PROCESSING = 'INITIAL_PROCESSING',
+    PERFORMING_OPERATIONS = 'PERFORMING_OPERATIONS',
+    SYNCHRONIZING = 'SYNCHRONIZING',
+    VALIDATING_OPERATIONS = 'VALIDATING_OPERATIONS',
+}
+
+export const SnapshotsReportStagesBinary = [
+    SnapshotsReportStages.INITIAL_PROCESSING,
+    SnapshotsReportStages.SYNCHRONIZING,
+    SnapshotsReportStages.COMPUTING_OPERATIONS,
+];
+
+export const SnapshotsReportStagesPercentage = [
+    SnapshotsReportStages.VALIDATING_OPERATIONS,
+    SnapshotsReportStages.PERFORMING_OPERATIONS,
+];
+
 export interface ResourceSnapshotsReportModel {
+    /**
+     * Current stage of the report
+     */
+    currentStage?: {
+        /**
+         * The progress value of the current step
+         * The value can be percentage type (from 0 to 100) binary type (0 or 1)
+         */
+        progressValue: number;
+        /**
+         * Current step name
+         */
+        stage: SnapshotsReportStages;
+    };
```

Comment:
@y-lakhdar progress bar possibility?

Reply:
@toofff how do you know when `progressValue` is a binary or percentage

Reply:
The API doesn't tell me, I had to add this to make it clear
```
export const SnapshotsReportStagesBinary = [
    SnapshotsReportStages.INITIAL_PROCESSING,
    SnapshotsReportStages.SYNCHRONIZING,
    SnapshotsReportStages.COMPUTING_OPERATIONS,
];

export const SnapshotsReportStagesPercentage = [
    SnapshotsReportStages.VALIDATING_OPERATIONS,
    SnapshotsReportStages.PERFORMING_OPERATIONS,
];
```

Reply:
Mmh... I don't think this is a good idea. Let's loop in @sbaillargeon

Reply:
This is how it works with snapshots in the AdminUI project currently. The best idea would be to have an info in addition to the API but I don't have it at the moment.